app/controllers/auth/AuthAction.scala

```diff
@@ -71,56 +72,60 @@ class AuthActionImpl @Inject()(
             }
             .asInstanceOf[Request[A]]
 
-          block(
-            AuthenticatedRequest[A](
-              Some(domain.Nino(trustedHelper.principalNino)),
-              None,
-              credentials,
-              confidenceLevel,
-              Some(UserName(Name(Some(trustedHelper.principalName), None))),
-              logins.previousLogin,
-              Some(trustedHelper),
-              profile,
-              trimmedRequest
-            )
-          )
-
-        case nino ~ Enrolments(enrolments) ~ Some(credentials) ~ confidenceLevel ~ name ~ logins ~ None ~ profile =>
-          val saEnrolment = enrolments.find(_.key == "IR-SA").flatMap { enrolment =>
-            enrolment.identifiers
-              .find(id => id.key == "UTR")
-              .map(key => SelfAssessmentEnrolment(SaUtr(key.value), SelfAssessmentStatus.fromString(enrolment.state)))
+          (
+            affinityGroup,
+            credentialStrength == CredentialStrength.strong,
+            confidenceLevel >= ConfidenceLevel.L200,
+            optionalHelper
+          ) match {
+            case (Individual, false, _, _) =>
+              upliftCredentialStrength
+            case (Individual, _, false, _) =>
+              upliftConfidenceLevel()(request)
+            case (Organisation | Agent, _, false, _) =>
+              upliftConfidenceLevel()(request)
+            case (Organisation | Agent, false, _, _) =>
+              upliftCredentialStrength
+            case (Individual, true, true, trustedHelper @ Some(helper)) =>
+              block(
+                AuthenticatedRequest[A](
+                  Some(domain.Nino(helper.principalNino)),
+                  None,
+                  credentials,
+                  confidenceLevel,
+                  Some(UserName(Name(Some(helper.principalName), None))),
+                  logins.previousLogin,
+                  trustedHelper,
+                  profile,
+                  trimmedRequest
+                )
+              )
+
+            case _ =>
+              val saEnrolment = enrolments.find(_.key == "IR-SA").flatMap { enrolment =>
```

Comment:
You could change the first find hof to include a value that indicates it is not a trusted helper then you wouldn't need two match statements to construct the authenticated request like so 
```suggestion
              val saEnrolment = enrolments.find(_.key == "IR-SA" && trustedHelper.nonEmpty).flatMap { enrolment =>
```